model_zoo/vilt/vilt.py

```diff
@@ -0,0 +1,227 @@
+from typing import Callable
+import torch
+from colossalai import nn as col_nn
+from colossalai.registry import MODELS
+from torch import dtype, nn
+from model_zoo.vit.vit import ViTBlock, ViTEmbedding
+import torch.nn.functional as F
+from colossalai.nn.layer.colossalai_layer import LayerNorm
+from transformers.models.bert.modeling_bert import BertConfig, BertEmbeddings, BertPredictionHeadTransform
+
+
+@MODELS.register_module
+class ViLT(nn.Module):
+    """
+    Vision Language Transformer
+    Capable for masked language modeling
+    """
+    def __init__(
+            self,
+            max_text_len: int,
+            num_layers: int,
+            vocab_size: int,
+            hidden_size: int,
+            img_size: int = 384,
+            patch_size: int = 16,
+            in_chans: int = 3,
+            depth: int = 12,
+            num_heads: int = 12,
+            dim: int = 768,
+            mlp_ratio: int = 4,
+            attention_dropout: float = 0.,
+            dropout: float = 0.1,
+            drop_path: float = 0.,
+            layernorm_epsilon: float = 1e-6,
+            activation: Callable = nn.functional.gelu,
+            dtype: dtype = None,
+            bias: bool = True,
+            checkpoint: bool = False,
+            init_method: str = 'torch',):
+
+        super().__init__()
+        max_sequence_length = max_text_len
+        num_layers = num_layers
+        vocab_size = vocab_size
+        self.vocab_size = vocab_size
+        hidden_size = hidden_size
+        self.num_layers = num_layers
+
+        bert_config = BertConfig(
+            vocab_size=vocab_size,
+            hidden_size=hidden_size,
+            num_hidden_layers=num_layers,
+            num_attention_heads=num_heads,
+            intermediate_size=hidden_size * mlp_ratio,
+            max_position_embeddings=max_sequence_length,
+            hidden_dropout_prob=dropout,
+            attention_probs_dropout_prob=attention_dropout,
+        )
+
+        self.pooler = Pooler(hidden_size)
+        self.token_type_embeddings = nn.Embedding(2, hidden_size)
+        self.token_type_embeddings.apply(init_weights)
+        self.text_embedding = BertEmbeddings(bert_config)
```

Comment:
If the model zoo is designed to run with all kinds of colossal model parallelism.
Should you implement a BertEmbedding using the `PatchEmbedding` from `from colossalai import nn as col_nn`?
https://github.com/hpcaitech/ColossalAI/blob/main/model_zoo/vit/vit.py#L65
@kurisusnowdeng @FrankLeeeee  any comments?

Reply:
Yes, I think so